code/modules/projectiles/guns/energy.dm

```diff
@@ -89,13 +98,114 @@ GLOBAL_LIST_INIT(registered_cyborg_weapons, list())
 					return suit.cell
 	return null
 
+/obj/item/weapon/gun/energy/special_check(var/mob/user)
+
+	if(!..())
+		return
+	
+	if(hatch_open)
+		to_chat(user, "<span class='warning'>[src] has its cell cover still open!</span>")
+		return 0
+	return 1
+
+//Attempts to load A into src, depending on the type of thing being loaded and the load_method
+//reloading in a new energy cell
+/obj/item/weapon/gun/energy/proc/load_ammo(var/obj/item/A, mob/user)
+	if(self_recharge)
+		return
+	//only let's you load in power cells
+	if(istype(A, /obj/item/weapon/cell))
+		. = TRUE
+		var/obj/item/weapon/cell/AM = A
+
+		if(hatch_open)
+			if(power_supply)
+				to_chat(user, "<span class='warning'>[src] already has a power cell loaded.</span>") //already a magazine here
+				return
+			if(AM.maxcharge <= (max_shots*charge_cost))
+				current_user = user
+				user.visible_message("[user] begins to reconfigure the wires and insert the cell into [src].","<span class='notice'>You begin to reconfigure the wires and insert the cell into [src].</span>")
+				
+				if(!do_after(user, reload_time, src))
+					user.visible_message("[usr] stops reconfiguring the wires in [src].","<span class='warning'>You stop reconfiguring the wires in [src].</span>")
+					return
+				if(!user.unEquip(AM, src))
+					return
+				power_supply = AM
+				user.visible_message("[user] inserts [AM] into [src].", "<span class='notice'>You insert [AM] and hot wire it into [src].</span>")
+				playsound(loc, mag_insert_sound, 50, 1)
+				AM.update_icon()
+			else
+				user.visible_message("<span class='warning'>The cell size is too big for the [src]. It must be [max_shots*charge_cost] Wh or smaller.</span>")
+				return
+		else
+			user.visible_message("<span class='warning'>The cell cover is closed. Use a screwdriver to open it.</span>")
+			return
+	update_icon()
```

Comment:
Probably meant to be in the actual cell code, hitting lasers with random things shouldn't need to update them.